src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SEPARATOR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(createReport(stringDataReport(fromFileName)),
+                new File(toFileName));
+    }
+
+    private String[] stringDataReport(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        String value;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((value = bufferedReader.readLine()) != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file" + fromFileName, e);
+        }
+        return stringBuilder.toString().split(System.lineSeparator());
+    }
+
+    private StringBuilder createReport(String[] text) {
+        int supply = 0;
+        int buy = 0;
+        int result;
+        String[] strings;
+        for (String s : text) {
+            strings = s.split(SEPARATOR);
+            if (strings[0].equals(SUPPLY)) {
+                supply += Integer.valueOf(strings[1]);
+            } else if (strings[0].equals(BUY)) {
+                buy += Integer.valueOf(strings[1]);
+            }
+        }
+        result = supply - buy;
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("supply,").append(supply).append(System.lineSeparator())
+                .append("buy,").append(buy).append(System.lineSeparator())
+                .append("result,").append(result).append(System.lineSeparator());
+        return stringBuilder;
```

Comment:
Return StringBuilder isn't good idea. You method createReport should do all logic that related to create some report, so the method should return completed result
return stringBuilder.toString();